src/bootstrap/tool.rs

```diff
@@ -772,7 +776,19 @@ impl Step for RustAnalyzerProcMacroSrv {
             extra_features: vec!["proc-macro-srv/sysroot-abi".to_owned()],
             is_optional_tool: false,
             source_type: SourceType::InTree,
-        })
+        })?;
+
+        // Copy `rust-analyzer-proc-macro-srv` to `build/triple/stageN/libexec/`
+        // so that r-a can use it.
+        let libexec_path = builder
+            .out
+            .join(&*builder.config.build.triple)
+            .join(format!("stage{}", self.compiler.stage))
+            .join("libexec");
```

Comment:
This is not correct for stage0 . Please use builder.sysroot(stage) instead, it has a bunch of logic that accounts for download-rustc as well.

Reply:
> `builder.sysroot(stage)`

Ah, so that's how you are supposed to do this